napari/components/_viewer_key_bindings.py

```diff
@@ -56,18 +55,21 @@ def toggle_ndisplay(viewer: Viewer):
 # ```
 # RuntimeError: wrapped C/C++ object of type CanvasBackendDesktop has been deleted
 # ```
-@register_viewer_action(trans._("Toggle theme."))
+@register_viewer_action(trans._("Toggle current viewer theme."))
 def toggle_theme(viewer: Viewer):
-    """Toggle theme for viewer"""
-    settings = get_settings()
+    """Toggle theme for current viewer"""
     themes = available_themes()
-    current_theme = settings.appearance.theme
+    current_theme = viewer.theme
+    # Check what the system theme is, to toggle properly
+    if current_theme == 'system':
+        current_theme = get_system_theme()
```

Comment:
it will be nice to test it as it could be easy broken in the future.

Reply:
Hmm, I thought I was testing this with the `loop through all `available_themes()` approach.
I guess `get_system_theme` isn't in any tests?
https://github.com/napari/napari/search?q=get_system_theme
Not sure how to test CI `system` theme.
Do i just try to monkeypatch `get_system_theme` to return say 'light` (aka not the default?) 
So I'd need to add a test in `_qt/_tests` for `get_system_theme` and then another test under `components/_tests/test_viewer_keybinds.py` for the toggle behavior from `'system'`, correct?

Reply:
`get_system_theme` may be tested, but I wrote about logic in this function.

Reply:
Ok, I think I have tests for both `get_system_theme` and the specific behavior of `toggle_theme`